src/index.tsx

```diff
@@ -10,13 +10,21 @@ import React, {
   useMemo,
   useRef,
   ComponentType,
+  ErrorInfo,
 } from "react";
 
 // eslint-disable-next-line @typescript-eslint/ban-types
 type ComponentDidCatch = ComponentLifecycle<{}, {}>["componentDidCatch"];
 
-interface ErrorBoundaryProps {
+type ErrorType = ErrorData | undefined;
+
+interface ErrorData {
   error: unknown;
```

Comment:
I chose `unknown` here as we can't guarantee the thrown value will be an instance of `Error`, and I think enforcing type safety is more important than convenience, if this is meant to be a back-up/fail-safe for other errors.

Using `instanceof Error` would be enough to please Typescript into allowing access to `Error`'s attributes.